sdk/core/azure-core/inc/azure/core/datetime.hpp

```diff
@@ -12,19 +12,161 @@
 #include <string>
 
 namespace Azure { namespace Core {
+
+  /**
+   * @brief An exception that gets thrown when @DateTime error occurs.
+   */
+  class DateTimeException : public std::runtime_error {
+  public:
+    /**
+     * @brief Construct with message string.
+     *
+     * @param message Message string.
+     */
+    explicit DateTimeException(std::string const& message) : std::runtime_error(message) {}
+  };
+
   /**
    * @brief Manages date and time in standardized string formats.
    */
   class DateTime {
   public:
-    /// A type that represents tick spans.
-    typedef uint64_t IntervalType;
-
-  private:
-    //  Number of seconds between 01-01-1970 and 01-01-1601.
-    static constexpr IntervalType WindowsToPosixOffsetSeconds = 11644473600LL;
+    /**
+     * @brief Represents time duration.
+     */
+    class Duration {
```

Comment:
Why is this class nested within DateTime rather than it being standalone?

Reply:
It does not have much use outside DateTime, they are coupled, and it is the same scope as prefious typedef uint64_t duration.

Reply:
I still question the nested-ness. I get that these frequently used together but this isn't mandatory. For example, we could have a function that accepts a timeout as a Duration (no DateTime is used).